tests/ArgentPonyWarcraftClient.Integration.Tests/GameDataApi/CovenantApiTests.cs

```diff
@@ -0,0 +1,143 @@
+                result,
+                "https://us.api.blizzard.com/data/wow/covenant/soulbind/9?namespace=static-us&locale=en_US"
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Conduits_Index_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<ConduitsIndex> result = await client.GetConduitsIndexAsync("static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/covenant/conduit/index?namespace=static-us&locale=en_US"
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Covenant_Media_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<CovenantMedia> result = await client.GetCovenantMediaAsync(3, "static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/media/covenant/3?namespace=static-us&locale=en_US"
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Covenants_Index_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<CovenantsIndex> result = await client.GetCovenantsIndexAsync("static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/covenant/index?namespace=static-us&locale=en_US"
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Soulbinds_Index_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<SoulbindsIndex> result = await client.GetSoulbindsIndexAsync("static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/covenant/soulbind/index?namespace=static-us&locale=en_US"
+            );
```

Comment:
Since so many of these tests perform the same checks on the `Success`, `Error`, etc., we might want to roll those into another helper method.

Reply:
Yup, that's my plan 😊 Focused on the "get it working" part this morning and gonna do the refinement next.